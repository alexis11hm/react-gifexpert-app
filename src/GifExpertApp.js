
import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = () =>{


    const [categories, setCategories] = useState(['Simpsons'])

    /*const handleAdd = () => {
        //Mantengo las categorias anteriores y agrego una nueva
        setCategories([...categories,'El pepe'])
        //setCategories( (cats) => [...categories,'el pepe']) //es lo mismo de arriba
    }*/
    return (
        <>
            <h2>GifExpert</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
            {
                <ol>
                {
                    categories.map((category) => (
                        <GifGrid 
                            key = {category}
                            category = {category} />
                    ))
                }
                </ol>
            }
        </>
    )

}