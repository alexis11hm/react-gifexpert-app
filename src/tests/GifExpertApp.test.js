import { shallow } from "enzyme"
import React from 'react'
import {GifExpertApp} from "../GifExpertApp"


describe('Pruebas en el <GifExpertAPP/>',()=>{


    test('se debe mostrar correctamente',()=>{
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
        }
    )

    test('debe de mostrar una lista de categorias',()=>{
        const categories = ['Simpsons','godzilla']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
        }
    )


})