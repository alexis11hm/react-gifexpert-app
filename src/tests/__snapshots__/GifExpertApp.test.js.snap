// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertAPP/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Simpsons"
      key="Simpsons"
    />
    <GifGrid
      category="godzilla"
      key="godzilla"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el <GifExpertAPP/> se debe mostrar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
